aiohttp/streams.py

```diff
@@ -307,11 +310,14 @@ async def _wait(self, func_name: str) -> None:
         try:
             if self._timer:
                 with self._timer:
-                    await waiter
+                    await asyncio.wait((waiter,))
             else:
-                await waiter
+                await asyncio.wait((waiter,))
         finally:
             self._waiter = None
+            fut_exc = waiter.exception()
+            if fut_exc:
+                raise fut_exc
```

Comment:
How is new version behavior different from original source?

Reply:
In original code `waiter.exception()` always empty, replacing `await waiter` with `await asyncio.wait((waiter,))` fix it, honestly i don't know why, i will try to read docs about `Future` more

Reply:
`waiter.exception()` was not called in the original code. It would always be `None` because an exception would already have been raised and would never reach this code.

These changes appear to do the same thing as the original code (wait for a future and raise any exceptions from it), but also break the code if `waiter` is not completed (due to the timeout), at which point it raises `InvalidStateError` as can be seen in the failing tests (in the previous test run, Github seems to be failing to run pytest on the latest commit).

Reply:
i added calling `waiter.exception()` and raising exception from `waiter` also removed  `raise self._exception` from `StreamReader.read` and can't catch any exception

Reply:
So as i can see `set_exception` here https://github.com/aio-libs/aiohttp/blob/15941aa1d941716d6fefe67f19369139bc3d0843/aiohttp/streams.py#L163
have no sense, because it never retrieved